static/sass/_patterns_media-object--small.scss

```diff
@@ -0,0 +1,15 @@
+@mixin p-media-object--small {
+  .p-media-object--small {
```

Comment:
I feel like we don't really need to introduce another pattern.
If all it's doing is making image smaller, we can just set image size in HTML.

Also forcing `__details` to have top margin makes it usable only with "centered" heading.

I believe same (or at least similar results) can be achieved with `.p-media-object`:
```html
<a class="p-media-object u-vertically-center" href="/install/{{package_name}}/arch">
    <img class="p-media-object__image" width="2.5rem" height="2.5rem" src="https://assets.ubuntu.com/v1/feca0fc0-Distro_Logo_ArchLinux.svg" alt="">
    <span class="p-media-object__details">
         <h4 class="p-media-object__title p-heading--five u-no-margin--bottom">Arch Linux</h4>
    </span>
</a>
```